README.adoc

```diff
@@ -6,6 +6,133 @@ Rocket implementation of the Application Container Specification.
+
+The stage1 image is an Application Container Specification compliant
+image file.  It contains the intermediate framework and the tools
+necessary to set up the pre-start environment for a new container.
+Some of the contents are compiled as part of the Rocket build
+process.  Others are standard OS startup files (systemd).  These are
+currently extracted from a stock CoreOS bootable image, but they could
+be built from source or pulled from other packages depending on the
+distribution they are pulled from.
+
+The `stage1.img` file is used by the `rkt` binary when a new container
+is run.  It can be replaced at the invocation using the
+`--stage1-image` option.
+
+The stage1 image is expected to be in the same directory as the `rkt`
+binary.  For packaging, the location of the stage1 image should be
+expanded to a search path.  This would allow distributions with
+specifications for the locations of extra binaries to conform.
+
+For RPM based packaging, the default stage1 image should reside in
+`/usr/libexec/rocket/stage1.img`.
+
+=== Runtime Storage and Database
+
+Rocket retrieves and unpacks Application Container images, and
+instantiates containers on behalf of users.  The container image files
+and the containers themselves must have a place on the host
+filesystem.
+
+The runtime storage and control databases reside in
+`/var/lib/rocket`. The directories are created and ownership and
+permission set when the package is installed.
+
+== Rocket Group, Ownership and Permissions
+
+Currently the enabling technology for Rocket, _nsenter_ requires root
+privileges. However, a number of precursor operations, such as
+management of image files and pre-run or post-run containers, do not
+require root.
+
+Non-root operations can be enabled by creating a _rocket_ group (in
+`/etc/group`) and setting the group ownership and permissions of the
+directories in `/var/lib/rocket` so that members of the rocket group
+have access.
+
+The Rocket package should create the _rocket_ group if it is not
+present at installation, and the package installation should set the
+group and permissions of the storage and database directories to
+enable non-root operations.
+
+== Sources
+
+The Rocket RPM will be built entirely from upstream sources. Any
+auxilliary files will be tracked here with the goal of getting them
+accepted upstream.
+
+=== Rocket Binary Sources
+
+The Rocket sources will be retrieved from the Github repository at or
+just prior to package build time. If they are retrieved prior to
+build, they will be collected in a gzipped tar archive for holding.
+
+=== Stage1 Image Sources
+
+The rocket deliverable requires a *stage1 image* which is the template
+used to create the namespaces which will contain the image processes.
+
+Currently the stage1 image is created using elements pulled from a
+CoreOS run-time image.  This runtime image is retrieved and then
+cached but it is treated as a binary "blob" whos provinence cannot be
+traced.
```

Comment:
Yes, if they have an existing mechanism which we can tweak, that would avoid problems of least-unexpected-behavior.